spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala

```diff
@@ -552,83 +562,88 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
 
   test("like (LikeSimplification enabled)") {
     val table = "names"
-    withTable(table) {
-      sql(s"create table $table(id int, name varchar(20)) using parquet")
-      sql(s"insert into $table values(1,'James Smith')")
-      sql(s"insert into $table values(2,'Michael Rose')")
-      sql(s"insert into $table values(3,'Robert Williams')")
-      sql(s"insert into $table values(4,'Rames Rose')")
-      sql(s"insert into $table values(5,'Rames rose')")
-
-      // Filter column having values 'Rames _ose', where any character matches for '_'
-      val query = sql(s"select id from $table where name like 'Rames _ose'")
-      checkAnswer(query, Row(4) :: Row(5) :: Nil)
-
-      // Filter rows that contains 'rose' in 'name' column
-      val queryContains = sql(s"select id from $table where name like '%rose%'")
-      checkAnswer(queryContains, Row(5) :: Nil)
-
-      // Filter rows that starts with 'R' following by any characters
-      val queryStartsWith = sql(s"select id from $table where name like 'R%'")
-      checkAnswer(queryStartsWith, Row(3) :: Row(4) :: Row(5) :: Nil)
-
-      // Filter rows that ends with 's' following by any characters
-      val queryEndsWith = sql(s"select id from $table where name like '%s'")
-      checkAnswer(queryEndsWith, Row(3) :: Nil)
+    withSQLConf(CometConf.COMET_CAST_ALLOW_INCOMPATIBLE.key -> "true") {
```

Comment:
I may need to remove some of these changes because I discovered that some of these tests were failing for me because of a "file already exists" error. I am looking into this now.